tests/ssg_test_suite/rule.py

```diff
@@ -303,89 +292,86 @@ def _slice_sbr(self, scenarios_by_rule_id, slice_current, slice_total):
                 new_sbr[rule_id] = [scenario]
         return new_sbr
 
-    def _test_target(self, target):
-        rules_to_test = self._get_rules_to_test(target)
-        if not rules_to_test:
-            logging.error("No tests found matching the rule ID(s) '{0}'".format(", ".join(target)))
-            return
+    def _get_all_rule_scenarios(self, rule):
+        product_yaml = common.get_product_context(self.test_env.product)
+        # Initialize a mock template_builder.
+        empty = "/ssgts/empty/placeholder"
+        template_builder = ssg.templates.Builder(
+            product_yaml, empty, common._SHARED_TEMPLATES, empty, empty)
+
+        # All tests is a mapping from path (in the tarball) to contents
+        # of the test case. This is necessary because later code (which
+        # attempts to parse headers from the test case) don't have easy
+        # access to templated content. By reading it and returning it
+        # here, we can save later code from having to understand the
+        # templating system.
+        all_tests = dict()
+
+        # Start by checking for templating tests and provision them if
+        # present.
+        templated_test_scenarios = common.fetch_templated_test_scenarios(
+            rule, template_builder, rule.directory, product_yaml,
+            rule.local_env_yaml)
+        all_tests.update(templated_test_scenarios)
+
+        # Add additional tests from the local rule directory. Note that,
+        # like the behavior in template_tests, this will overwrite any
+        # templated tests with the same file name.
+        local_test_scenarios = common.fetch_local_test_scenarios(
+            rule.directory, rule.local_env_yaml)
+        all_tests.update(local_test_scenarios)
+
+        # Filter out everything except the shell test scenarios.
+        # Other files in rule directories are editor swap files
+        # or other content than a test case.
+        allowed_scripts = filter(lambda x: x.endswith(".sh"), all_tests)
+        content_mapping = {x: all_tests[x] for x in allowed_scripts}
 
+        scenarios = []
+        for script, script_contents in content_mapping.items():
+            scenario = Scenario(script, script_contents)
+            scenarios.append(scenario)
+        return scenarios
+
+    def _filter_rule_scenarios(self, rule_scenarios):
+        scenarios = []
+        for scenario in rule_scenarios:
+            scenario.override_profile(self.scenarios_profile)
+            if (scenario.matches_regex(self.scenarios_regex) and
+                    scenario.matches_platform(self.benchmark_cpes)):
+                scenarios.append(scenario)
+        return scenarios
```

Comment:
This says that it filters rule scenarios, but in fact some "filtering" already happened elsewhere, specifically filtering of the templated test scenarios based on the test_config.yml happened in `common.fetch_templated_test_scenarios()` , but that's called from of `_get_all_rule_scenarios`.

So from one point of view it's misleading because:
- `_get_all_rule_scenarios` doesn't get all test scenarios but it gets only those that are allowed by test_config.yml.
- `filter_rule_scenarios` doesn't perform filtering test scenarios base on `test_config.yml` but only based on profile, regex and platform CPE (in other words only based on command line arguments provided by user).

But from the second point of view it's fine because:
- the feature that local test scenarios override templated test scenarios with the same name is now implemented in an easy and elegant way (simply updating the keys within dictionary)
- code handling templated test scenarios is decoupled from code handling not templated test scenarios

Reply:
@matejak Do you have any opinion on this subject? It's a question of logical decoupling of responsibilities between different parts of code.